budget_control/models/budget_period.py

```diff
@@ -324,6 +324,21 @@ def _prepare_matrix_by_analytic(self, instance, analytic_ids):
     def _prepare_matrix_all_analytics(self, instance, analytic_ids):
         """Return resulting matrix of all analytic combined."""
         ctx = {"filter_analytic_ids": analytic_ids}
+        filter_activity_group = self._context.get(
```

Comment:
Why not just use one line,
```
ctx.update(self.env.context)
```

Or if you worry about others context value, you can use normal python filter() first to get only dict with key start with `filter_` first. Like (I didn't test but you understand the concept,

```
filter_ctx = filter(lambda k, v: 'filter_' in k, self.env.context)
ctx.update(filter_ctx)
```

I think the main concept of what I always say is simply, 1) Reuse code (try to not write more code for doing the same thing 2) Make it as short as possible.

This will reduce bugs and make code more maintainable.

Reply:
Fixed, Thank you for your reviewed.
I think this code is easy and correct.

```
ctx = self._context.copy()
ctx.update({"filter_analytic_ids": analytic_ids})
```

Reply:
You can merge when ready.